packages/memory-inspector/src/browser/memory-widget/memory-options-widget.tsx

```diff
@@ -0,0 +1,726 @@
+                        title={nls.localize('theia/memory-inspector/lengthTooltip', 'Number of bytes to fetch, in decimal or hexadecimal')}
+                        defaultValue={this.readLength.toString()}
+                        passRef={this.assignReadLengthRef}
+                        onChange={Utils.validateNumericalInputs}
+                        onKeyDown={this.doRefresh}
+                        disabled={!this.doUpdateAutomatically}
+                    />
+                    <button
+                        type='button'
+                        className='theia-button main view-group-go-button'
+                        onClick={this.doRefresh}
+                        disabled={!this.doUpdateAutomatically}
+                        title={nls.localizeByDefault('Go')}
+                    >
+                        {nls.localizeByDefault('Go')}
+                    </button>
+                </div>
+                <div className={`t-mv-memory-fetch-error${this.showMemoryError ? ' show' : ' hide'}`}>
+                    {this.memoryReadError}
+                </div>
+            </>
+        );
+    }
+
+    protected activateHeaderInputField = (e: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>): void => {
+        if (!this.isTitleEditable) {
+            const isMouseDown = !('key' in e);
+            const isActivationKey = 'key' in e && (
+                KeyCode.createKeyCode(e.nativeEvent).key?.keyCode === Key.SPACE.keyCode
+                || KeyCode.createKeyCode(e.nativeEvent).key?.keyCode === Key.ENTER.keyCode
+            );
+            if (isMouseDown || isActivationKey) {
+                if (isMouseDown) {
+                    e.currentTarget.blur();
+                }
+                this.isTitleEditable = true;
+                this.update();
+            }
+        }
+    };
+
+    protected saveHeaderInputValue = (e: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>): void => {
+        const isMouseDown = !('key' in e);
+        const isSaveKey = 'key' in e && e.key === 'Enter';
+        const isCancelKey = 'key' in e && e.key === 'Escape';
+        e.stopPropagation();
+        if (isMouseDown || isSaveKey || isCancelKey) {
+            if (!isCancelKey && this.headerInputField) {
+                this.title.label = this.headerInputField.value;
+                this.title.caption = this.headerInputField.value;
+            }
+
+            this.isTitleEditable = false;
+            this.update();
+        }
+    };
+
+    protected assignHeaderInputRef = (element: HTMLInputElement): void => {
+        if (element) {
+            this.headerInputField = element;
+            element.focus();
+        }
+    };
+
+    protected renderToolbar(): React.ReactNode {
+        return (
+            <div className='memory-widget-toolbar'>
+                {this.memoryWidgetOptions.dynamic !== false && (
```

Comment:
It would be really great if this could also be moved into a separate method to support easier overriding also allow the user to add something to the toolbar before the lock.

Reply:
Done.